diff --git a/node_modules/react-native-reanimated/ios/native/NativeMethods.mm b/node_modules/react-native-reanimated/ios/native/NativeMethods.mm
index 58f1e20..505dd43 100644
--- a/node_modules/react-native-reanimated/ios/native/NativeMethods.mm
+++ b/node_modules/react-native-reanimated/ios/native/NativeMethods.mm
@@ -37,14 +37,10 @@
 }
 
 
-NSString *eventDispatcherKey = @"eventDispatcher";
 void scrollTo(int scrollViewTag, RCTUIManager *uiManager, double x, double y, bool animated) {
   UIView *view = [uiManager viewForReactTag:@(scrollViewTag)];
   RCTScrollView *scrollView = (RCTScrollView *) view;
-  RCTEventDispatcher* oldEventDispatcher = [scrollView valueForKey:eventDispatcherKey];
-  [scrollView setValue:nil forKey:eventDispatcherKey];
   [scrollView scrollToOffset:(CGPoint){(CGFloat)x, (CGFloat)y} animated:animated];
-  [scrollView setValue:oldEventDispatcher forKey:eventDispatcherKey];
 }
 
 }
